### ISSUE TITLE
False positive on unused_optional_binding rule for if-let syntax when ignore_optional_try is set to true

### ISSUE THREAD

**Issue by jazminebarroga** [realm/SwiftLint#3220]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `ignore_optional_try` option, when set to true, for the `unused_optional_binding` rule triggers a violation for the if-let syntax such as: 
`if let _ = try? alwaysThrow() `

but works as expected when 
`guard let _ = try? alwaysThrow()`

Both lines of code have the optional try so I don't see why the if-let syntax would trigger a violation

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint --config swiftlint.yml
 warning: Unused Optional Binding Violation: Prefer `!= nil` over `let _ =` (unused_optional_binding)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.39.2
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Cocoapods
* Paste your configuration file:

```yml
whitelist_rules: 
  - unused_optional_binding

unused_optional_binding:
  ignore_optional_try: true
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)? 11.3

```swift
// This triggers a violation:
if let _ = try? alwaysThrow() 
```

**Comment by marcelofabri** [comment]
IIRC this was intentional, but not sure why. Feel free to submit a PR updating the behavior!
